common/transport/amqp/src/receiver_link.ts

```diff
@@ -21,57 +28,119 @@ const debug = dbg('azure-iot-amqp-base:ReceiverLink');
 /*Codes_SRS_NODE_AMQP_RECEIVER_LINK_16_003: [** The `ReceiverLink` class shall implement the `AmqpLink` interface.]*/
 export class ReceiverLink  extends EventEmitter implements AmqpLink {
   private _linkAddress: string;
-  private _linkOptions: any;
-  private _linkObject: any;
+  private _linkOptions: ReceiverOptions;
+  private _rheaReceiver: Receiver;
   private _fsm: machina.Fsm;
-  private _amqp10Client: any;
-  private _detachHandler: (detachEvent: any) => void;
-  private _errorHandler: (err: Error) => void;
-  private _messageHandler: (message: AmqpMessage) => void;
+  private _rheaSession: Session;
+  //
+  // We want to know if a detach has already occurred, having been sent by the peer.  We will want to send a matching
+  // detach but we can't expect a reply to it.
+  //
+  private _receiverCloseOccurred: boolean = false;
+  //
+  // The following to fields contain the callbacks that were sent in to request the attach/detach.  Since
+  // the work will be completed by event listeners, we needed a place to keep the callback for access
+  // within the handler.
+  //
+  private _attachingCallback: (err: Error | AmqpError, result?: any) => void;
+  private _detachingCallback: (err: Error | AmqpError, result?: any) => void;
+  //
+  // Used to hold onto an error that was indicated by a receiver_error event.  The error will be utilized
+  // by the receiver_close code when it transitioning and calling back to the owner of the ReceiverLink.
+  //
+  private _indicatedError: Error | AmqpError;
+  //
+  // Create a name to uniquely identify the link.  Used for debugging purposes now.  If necessary it could
+  // be used to distinguish receivers.
+  //
+  private _rheaReceiverName: string;
+  private _combinedOptions: ReceiverOptions;
+  //
+  // When we receive a message from the service we send it up to the application.  We have no say as to how we
+  // dispose of the message.
+  //
+  // We need for the application to indicate the disposition.
+  //
+  // We create a dictionary that is keyed by the amqpMessage.  The value of the dictionary is the delivery record associated
+  // (by rhea) of this message.
+  //
+  // So when a message is received we place in the dictionary.  When the application sends back down the message we look it up
```

Comment:
>sends back down the message [](start = 85, length = 27)

you mean "when the user of the SDK settles the message by accepting, abandoning or rejecting it"?